research/mark_generator/training_data_generator.py

```diff
@@ -0,0 +1,77 @@
+import random
+import typing as tp
+from pathlib import Path
+
+import cv2
+import numpy as np
+
+from research.data_augmentator import DataAugmentator
+from research.mark_generator import MarkGenerator, Mark
+
+
+class TrainingDataGenerator:
+    ANSWER_BOXES_PATH = 'data/input/answer_boxes.jpg'
+    INDEX_BOXES_PATH = 'data/input/index_boxes.jpg'
+    OUTPUT_DIR = 'data/output'
+
+    def __init__(self, shape: tp.Tuple = (90, 90), box_offset: int = 4):
+        self._border = 5
+        self._shape = shape
+        self._box_offset = box_offset
+        self._raw_answer_boxes = self.extract_boxes()
+        self._mark_generator = self.set_mark_generator()
+
+    def set_mark_generator(self):
+        mark, unmark = Mark.get_random_valid_marks()
```

Comment:
Does that mean only one marking method can be generated at once? 
Is it possible to have a single `MarkGenerator` that generates multiple marking methods?